ts/src/binance.ts

```diff
@@ -6075,42 +6076,73 @@ export default class binance extends Exchange {
          * @returns {object[]} a list of [order structures]{@link https://docs.ccxt.com/#/?id=order-structure}
          */
         if (symbol === undefined) {
-            throw new ArgumentsRequired (this.id + ' cancelOrder() requires a symbol argument');
+            throw new ArgumentsRequired (this.id + ' cancelAllOrders() requires a symbol argument');
         }
         await this.loadMarkets ();
         const market = this.market (symbol);
         const request = {
             'symbol': market['id'],
         };
+        let isPortfolioMargin = undefined;
+        [ isPortfolioMargin, params ] = this.handleOptionAndParams2 (params, 'cancelAllOrders', 'papi', 'portfolioMargin', false);
+        const isConditional = this.safeBool2 (params, 'stop', 'conditional');
         const type = this.safeString (params, 'type', market['type']);
-        params = this.omit (params, [ 'type' ]);
-        const [ marginMode, query ] = this.handleMarginModeAndParams ('cancelAllOrders', params);
+        params = this.omit (params, [ 'type', 'stop', 'conditional' ]);
+        let marginMode = undefined;
+        [ marginMode, params ] = this.handleMarginModeAndParams ('cancelAllOrders', params);
         let response = undefined;
         if (market['option']) {
-            response = await this.eapiPrivateDeleteAllOpenOrders (this.extend (request, query));
+            response = await this.eapiPrivateDeleteAllOpenOrders (this.extend (request, params));
         } else if (market['linear']) {
-            response = await this.fapiPrivateDeleteAllOpenOrders (this.extend (request, query));
+            if (isPortfolioMargin) {
+                if (isConditional) {
+                    response = await this.papiDeleteUmConditionalAllOpenOrders (this.extend (request, params));
+                } else {
+                    response = await this.papiDeleteUmAllOpenOrders (this.extend (request, params));
+                }
+            } else {
+                response = await this.fapiPrivateDeleteAllOpenOrders (this.extend (request, params));
+            }
         } else if (market['inverse']) {
-            response = await this.dapiPrivateDeleteAllOpenOrders (this.extend (request, query));
+            if (isPortfolioMargin) {
+                if (isConditional) {
+                    response = await this.papiDeleteCmConditionalAllOpenOrders (this.extend (request, params));
+                } else {
+                    response = await this.papiDeleteCmAllOpenOrders (this.extend (request, params));
+                }
+            } else {
+                response = await this.dapiPrivateDeleteAllOpenOrders (this.extend (request, params));
+            }
         } else if ((type === 'margin') || (marginMode !== undefined)) {
-            if (marginMode === 'isolated') {
-                request['isIsolated'] = true;
+            if (isPortfolioMargin) {
```

Comment:
if you provide `portfolioMargin` I don't think you need to provide `marginMode` as well